### PR TITLE
Refactor upgrade test

### PR DESCRIPTION
**Pull request by daniel-butler-irl**
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Comment by daniel-butler-irl** [comment]
Working locally, I'm just not 100% sure how it will act in the pipelines. I hope it will "just work", but may need to set some extra variables for the git clone

File: testhelper/test_options.go [hunk 1]
```diff
@@ -71,6 +71,22 @@ type TestOptions struct {
...
+
+	// BaseGitPersonalAccessToken specifies the Personal Access Token (PAT) to use for Git operations.
+	// This token provides authentication the base repository is private and accessed over HTTPS.
+	// If the environment variable GIT_PAT is set, it will override this value.
+	BaseGitPersonalAccessToken string
+
+	// BaseGitPrivateSshKey specifies the private SSH key to use for Git operations.
+	// This key provides authentication when the repository is private and accessed over SSH.
+	// If the environment variable SSH_PRIVATE_KEY is set, it will override this value.
+	BaseGitPrivateSshKey string
```
Comment (toddgiguere) [inline testhelper/test_options.go:88]: This is incredibly risky for a user. I personally would never set either PAT or a private key in code. You could eliminate these two and just rely on ENV variables? would be a lot safer I think.

**Review by toddgiguere — CHANGES_REQUESTED** [review]

**Pull request merged**